### ISSUE TITLE
azurerm_azuread_application's default value for the homepage field is no longer valid

### ISSUE THREAD

**Issue by robinkb** [terraform-providers/terraform-provider-azurerm#1953]
As the documentation states, the `azurerm_azuread_application` resource defaults the `homepage` field to `http://[name]`. This used to work fine, but as of today, it seems that the Azure API no longer accepts the `http` scheme. I can work around the bug by providing the `homepage` field myself with the `https` scheme.

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

Terraform v0.11.8
+ provider.azurerm v1.15.0
+ provider.kubernetes v1.1.0
+ provider.local v1.1.0
+ provider.null v1.0.0
+ provider.random v1.3.1

### Affected Resource(s)

* azurerm_azuread_application

### Terraform Configuration Files

```hcl
resource "azurerm_azuread_application" "kubernetes" {
  name = "dev-kubernetes"
}
```

### Expected Behavior

Terraform creates an Azure AD Application.

### Actual Behavior

Terraform returned an error from the Azure API

```
azurerm_azuread_application.kubernetes: graphrbac.ApplicationsClient#Create: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="Unknown" Message="Unknown service error" Details=[{"odata.error":{"code":"Request_BadRequest","date":"2018-09-20T06:33:08","message":{"lang":"en","value":"Url is malformed paramName: WwwHomepage, paramValue: dev-kubernetes, objectType: Microsoft.Online.DirectoryServices.Application"},"requestId":"1ab56632-e57b-476b-82a1-0cb0bd88c111","values":[{"item":"PropertyName","value":"homepage"},{"item":"PropertyErrorCode","value":"NotAWellFormedUrl"}]}}]
```

I can work around the issue by specifying the `homepage` field with the `https` scheme.

```hcl
resource "azurerm_azuread_application" "kubernetes" {
  name     = "dev-kubernetes"
  homepage = "https://dev-kubernetes"
}
```

### Steps to Reproduce

Simply try to create an `azurerm_azuread_application` resource by providing only the name.

1. `terraform apply`

**Comment by katbyte** [comment]
Hey @robinkb,

Thank you for bringing this to our attention. I have opened #1960 to update the auto generated homepage & documentation. This should go out in 1.16 in the next couple weeks.

**Comment by robinkb** [comment]
Great! Thank you for fixing the bug.

**Comment by tombuildsstuff** [comment]
hey @robinkb 

Just to let you know that this fix has now landed in v1.16 of the AzureRM Provider (here's the full changelog](https://github.com/terraform-providers/terraform-provider-azurerm/blob/v1.16.0/CHANGELOG.md)). You can upgrade to to this version by specifying it in your provider block:

```
provider "azurerm" {
  version = "=1.16.0"
}
```

... and then running `terraform init -upgrade` :)

Thanks!
